### PR TITLE
composer 1.6.3 (new formula)

### PR DESCRIPTION
**Pull request by javian**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This is a migration from homebrew-php https://github.com/Homebrew/homebrew-php/pull/4796 .
Compared to the formula in the php tap this is easier to read but should accomplish the same thing.

### CODE REVIEW

**Comment by javian** [comment]
Reading up on the new script there might be some caveats around how its been written looking back at the related PRs.

**Comment by javian** [comment]
The concerns that were raised previously seems to have been unfounded so I would run with it as it is now.

**Comment by javian** [comment]
Just to reiterate my findings in here as well. The two settings that are mandated in the wrapper script are 1. Default in the homebrew Formual 2. One of them is not used. So to not complicate things I would just drop the phar in the bin directory and run it directly rather than messing with the wrapper script.

Also, composer has a self-update functionality where it will overwrite itself with the new version and as far as I can see there's no way to turn that off https://getcomposer.org/doc/03-cli.md#self-update-selfupdate-

File: formula/composer.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Composer < Formula
+  desc "Dependency Manager for PHP"
+  homepage "https://getcomposer.org"
+  url "https://getcomposer.org/download/1.6.3/composer.phar"
+  sha256 "52cb7bbbaee720471e3b34c8ae6db53a38f0b759c06078a80080db739e4dcab6"
+  revision 1
```
Comment (SMillerDev) [inline formula/composer.rb]: Do we need this?

**Review by SMillerDev — APPROVED** [review]

File: formula/composer.rb (hunk 1 above)
Comment (javian) [reply formula/composer.rb]: Not necessarily no but @ilovezfs added it to the php formula when it was migrated to force an upgrade so I assumed there was a good reason for that. If there isn't for this Formula we can remove it.

**Review by SMillerDev — APPROVED** [review]

File: formula/composer.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "52cb7bbbaee720471e3b34c8ae6db53a38f0b759c06078a80080db739e4dcab6"
+
+  bottle :unneeded
+
+  def install
+    bin.install "composer.phar" => "composer"
+  end
+
+  test do
+    system "#{bin}/composer", "--version"
```
Comment (fxcoudert) [inline formula/composer.rb]: We'll need a test that exercices some of the software's functionality…

**Review by fxcoudert — DISMISSED** [review]

File: formula/composer.rb (hunk 2 above)
Comment (javian) [reply formula/composer.rb]: @SMillerDev any ideas on this one ?

Comment (javian) [reply formula/composer.rb]: I'm thinking something like this should do it https://www.frobiovox.com/posts/2016/08/16/basic-hello-world-with-composer-and-php.html

**Comment by javian** [comment]
@fxcoudert the issue about self-update also exists in composer (just as you point out in the wp-cli formula) but this has been discussed with @ilovezfs and @MikeMcQuaid on slack and it was concluded that it would be ok in this case but I'll let @ilovezfs confirm that.

**Comment by ilovezfs** [comment]
yes

**Comment by javian** [comment]
@fxcoudert anything else you need for this to pass ?

File: formula/composer.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      // Autoload files using the Composer autoloader.
+      require_once __DIR__ . '/../vendor/autoload.php';
+
+      use HelloWorld\\Greetings;
+
+      echo Greetings::sayHelloWorld();
+    EOS
+
+    cd "test"
```
Comment (ilovezfs) [inline formula/composer.rb]: Is a `test` subdirectory really needed?

File: formula/composer.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    system "#{bin}/composer", "install"
+    assert_match /^HelloHomebrew$/, shell_output("php tests/test.php"), "Output does not match testcase"
```
Comment (ilovezfs) [inline formula/composer.rb]: The third argument is uneeded

File: formula/composer.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+class Composer < Formula
+  desc "Dependency Manager for PHP"
+  homepage "https://getcomposer.org"
```
Comment (ilovezfs) [inline formula/composer.rb]: trailing slash

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks again @javian!

**Pull request merged**